stl/inc/ranges

```diff
@@ -4749,6 +4752,400 @@ namespace ranges {
+                        if (_STD get<0>(_Inner_it) != _RANGES end(_Parent->_Pattern)) {
+                            break;
+                        }
+
+                        auto&& _Inner = _Update_inner();
+                        _Inner_it.template emplace<1>(_RANGES begin(_Inner));
+                    } else {
+                        auto&& _Inner = _Get_inner();
+                        if (_STD get<1>(_Inner_it) != _RANGES end(_Inner)) {
+                            break;
+                        }
+
+                        ++_Outer_it;
+                        if (_Outer_it == _RANGES end(_Parent->_Range)) {
+                            if constexpr (_Deref_is_glvalue) {
+                                _Inner_it.template emplace<0>();
+                            }
+                            break;
+                        }
+
+                        _Inner_it.template emplace<0>(_RANGES begin(_Parent->_Pattern));
+                    }
+                }
+            }
+
+        public:
+            using iterator_concept = //
+                conditional_t<_Deref_is_glvalue && bidirectional_range<_Base> //
+                                  && _Bidirectional_common<_InnerRng<_Const>> && _Bidirectional_common<_PatternBase>,
+                    bidirectional_iterator_tag,
+                    conditional_t<_Deref_is_glvalue && forward_range<_Base> && forward_range<_InnerRng<_Const>>,
+                        forward_iterator_tag, input_iterator_tag>>;
+            using value_type      = common_type_t<iter_value_t<_InnerIter>, iter_value_t<_PatternIter>>;
+            using difference_type = common_type_t<iter_difference_t<_OuterIter>, iter_difference_t<_InnerIter>,
+                iter_difference_t<_PatternIter>>;
+
+            // clang-format off
+            _Iterator() requires default_initializable<_OuterIter> = default;
+            // clang-format on
+
+            constexpr _Iterator(_Iterator<!_Const> _It) requires _Const //
+                && convertible_to<iterator_t<_Vw>, _OuterIter> //
+                && convertible_to<iterator_t<_InnerRng<false>>, _InnerIter> //
+                && convertible_to<iterator_t<_Pat>, _OuterIter> //
+                : _Outer_it{_STD move(_It._Outer_it)}, _Parent{_It._Parent} {
+                if (_It._Inner_it.index() == 0) {
+                    _Inner_it.template emplace<0>(_STD get<0>(_STD move(_It._Inner_it)));
+                } else {
+                    _Inner_it.template emplace<1>(_STD get<1>(_STD move(_It._Inner_it)));
+                }
+            }
+
+            _NODISCARD constexpr decltype(auto) operator*() const {
+                using _Ref = common_reference_t<iter_reference_t<_InnerIter>, iter_reference_t<_PatternIter>>;
+                return _STD visit([](auto&& _It) -> _Ref { return *_It; }, _Inner_it);
+            }
+
+            constexpr _Iterator& operator++() {
+                _STD visit([](auto&& _It) { ++_It; }, _Inner_it);

```

Comment:
We can directly grab into `_Variantish`. So we should be able to do something along those lines
```suggestion
                if (_Inner_it._Contains == _Variantish_state._Holds_iter) {
                    ++_Inner_it._Iterator;
                } else {
                    ++_Inner_it._Sentinel;    
                }

```